src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -178,6 +204,8 @@ private void ReceiveDocuments(DocumentsOperationContext context, BlittableJsonRe
 
 			if (changeVectorUpdated)
 				_database.DocumentsStorage.SetChangeVector(context, dbChangeVector);
+
+			return dbChangeVector.FirstOrDefault(x => x.DbId == Guid.Parse(_incomingMessageHeader.SourceDatabaseId)).Etag;
```

Comment:
What are you doing here? You are forcing us to parse the SourceDatabaseId for each item in the chagne vector.